File: app/src/main/java/vergecurrency/vergewallet/service/repository/ApifyService.java
package vergecurrency.vergewallet.service.repository;

import org.json.JSONException;
import org.json.JSONObject;

public class ApifyService {

	//TODO : Query should be done here
	//TODO : Make it a static class
	public ApifyService() {
	}

	//to be moved into a apify parser
	public String readIP(String rawData) {

		if (rawData != null && !rawData.equals("")) {
			JSONObject reader;
			try {
				reader = new JSONObject(rawData);
				return reader.getString("ip");
			} catch (JSONException e) {
				e.printStackTrace();
				return "error";
			}
		} else return "error";
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/service/repository/ApifyService.kt
package vergecurrency.vergewallet.service.repository

import cz.msebera.android.httpclient.client.methods.HttpGet
import org.json.JSONException
import org.json.JSONObject
import vergecurrency.vergewallet.Constants
import vergecurrency.vergewallet.service.model.network.layers.NetworkGateway

//TODO : Query should be done here
object ApifyService {

    //to be moved into a apify parser
    fun requestIP(): String {

        var rawData: String = NetworkGateway().doRequest(HttpGet(Constants.IP_RETRIEVAL_ENDPOINT))

        if (rawData != "") {
            val reader: JSONObject
            try {
                reader = JSONObject(rawData)
                return reader.getString("ip")
            } catch (e: JSONException) {
                e.printStackTrace()
                return "error"
            }

        } else
            return "error"
    }
}
